require('express-async-errors')
const express=require('express')
const mongoose=require('mongoose')
const notesRouter=require('./controllers/notes')
const usersRouter=require('./controllers/users')
const loginRouter=require('./controllers/login')
const config=require('./utils/config')
const logger=require('./utils/logger')
const middleware=require('./utils/middleware')
const app=express()
logger.info('connecting of url...')

mongoose.connect(config.MONGODB_URI)
.then(()=>{
    logger.info('connected successully')
})
.catch((error)=>{
    logger.error('error connecting to database')
})
app.use(express.static('dist'))
app.use(express.json())
app.use(middleware.requestLogger)
app.use('/api/notes',notesRouter)
app.use('/api/users',usersRouter)
app.use('/api/login',loginRouter)
app.use(middleware.unknownendpoint)
app.use(middleware.errorhandler)
module.exports=app
